packages/central-server/app/tasks/PatientMergeMaintainer.js

```diff
@@ -86,33 +88,58 @@ export class PatientMergeMaintainer extends ScheduledTask {
     );
   }
 
-  async remergePatientRecords() {
-    // set up an object for counting affected records
-    const counts = {};
-    const updateCounts = (name, records) => {
-      const len = records && records.length;
-      if (len) {
-        counts[name] = len;
-      }
-    };
+  async updateDependentRecordsForResync(merges) {
+    const encounters = merges['Encounter'] || [];
+    await refreshMultiChildRecordsForSync(this.models.Encounter, encounters);
 
-    // do all the simple model updates
-    for (const modelName of simpleUpdateModels) {
-      const model = this.models[modelName];
-      const records = await this.mergeAllRecordsForModel(model);
-      updateCounts(modelName, records);
-    }
+    // Patient Care Plans
+    const patientCarePlans = merges['PatientCarePlan'] || [];
+    await refreshMultiChildRecordsForSync(this.models.PatientCarePlan, patientCarePlans);
 
-    // then the model updates that need specific updates:
-    for (const modelName of specificUpdateModels) {
-      const method = this[`specificUpdate_${modelName}`];
-      if (method) {
-        const records = await method.call(this);
+    // Patient Death Data
+    const patientDeathDataRecords = merges['PatientDeathData'] || [];
+    await refreshMultiChildRecordsForSync(this.models.PatientDeathData, patientDeathDataRecords);
+  }
+
+  async remergePatientRecords() {
```

Comment:
This PR has the extra changes for `PatientMergeMaintainer` that we agreed to pull out of the hotfix as it contains too much regression risks, discussed with @edmofro [here](https://github.com/beyondessential/tamanu/pull/7829#pullrequestreview-2930594473)